pkgs/development/tools/java/jprofiler/default.nix

```diff
@@ -1,22 +1,70 @@
 { stdenv
 , lib
+, callPackages
 , fetchurl
 , makeWrapper
 , makeDesktopItem
 , copyDesktopItems
-, undmg
+, autoPatchelfHook
+, callPackage
 , jdk
+, libX11
+, musl
+, component ? "ui"
```

Comment:
We should add an assert that this is only ui or agent.

Reply:
done